subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/FileDependencyResolveIntegrationTest.groovy

```diff
@@ -176,6 +176,7 @@ class FileDependencyResolveIntegrationTest extends AbstractDependencyResolutionT
 '''
 
         when:
+        executer.expectDocumentedDeprecationWarning("""The resolved configuration 'compile' has been consumed as a variant, resulting in a circular dependency graph. Depending on the resolved configuration in this manner has been deprecated. This will fail with an error in Gradle 9.0. Be sure to mark configurations meant for resolution as canBeConsumed=false, or use the 'resolvable(String)' configuration factory method. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#depending_on_root_configuration""")
```

Comment:
🤔 I haven't looked through all the tests, but can you see if some of the tests actually rely on consuming a resolvable configuration? If they don't care, instead of adding an expected deprecation warning, we should change the test to not emit one.

Reply:
All of the tests that this deprecation affected specifically test some sort of behavior where there is a cycle in the graph